block/src/qcow/mod.rs

```diff
@@ -1571,11 +1576,25 @@ impl QcowFile {
         // Push L1 table and refcount table last as all the clusters they point to are now
         // guaranteed to be valid.
         let mut sync_required = if self.l1_table.dirty() {
-            self.raw_file.write_pointer_table(
-                self.header.l1_table_offset,
-                self.l1_table.get_values(),
-                0,
-            )?;
+            // Build L1 table with OFLAG_COPIED bits set correctly based on L2 cluster refcounts
+            let l1_active: Vec<u64> = self
```

Comment:
This creates a temporary vector. This is going to introduce a performance hit.

Why not create the correct `l1_table` from the get-go?

Reply:
I'm fine with leaving this optimization to a later date when it actually causes problems.

Even if we want to keep the code structure as-is, there is maybe still a way to avoid the temporary vector -- make `write_l1_table` takes an `impl iterator` instead of a slice.

Reply:
Ack, I was just checking to rewrite `write_pointer_table` to accept an iterator and a callback, so then that underlying method would query a refcount and imediately write an L1 entry. I'm diggging on this.

Thanks